wrappers/react/src/hotTableContext.tsx

```diff
@@ -123,28 +123,31 @@ const HotTableContextProvider: React.FC<PropsWithChildren> = ({ children }) => {
   }, [])
 
   const componentRendererColumns = useRef<Map<number | 'global', boolean>>(new Map());
+
   const editorCache = useRef<HotEditorCache>(new Map());
   const renderedCellCache = useRef<Map<string, HTMLTableCellElement>>(new Map());
   const clearRenderedCellCache = useCallback(() => renderedCellCache.current.clear(), []);
   const portalCacheArray = useRef<React.ReactPortal[]>([]);
 
-  const getRendererWrapper = useCallback((rendererElement: React.ReactElement): typeof Handsontable.renderers.BaseRenderer => {
+  const getRendererWrapper = useCallback((Renderer: React.ComponentType<HotRendererProps>): typeof Handsontable.renderers.BaseRenderer => {
```

Comment:
I do not know if this implementation would have the flicker issue the [pull request](https://github.com/handsontable/handsontable/pull/10758/files) resolves but it would be worth adding the guid at least coming from the instance
```
      // Handsontable.Core type is mising guid
      const instanceGuid = (instance as unknown as { guid: string }).guid;
      const portalContainerKey = `${instanceGuid}${key}`
      const portalKey = `${key}-${instanceGuid}`
```

in this pr the portal container is being cached which resolves a flickering problem, its really noticeable if you have a custom cell renderer with a background color.

Reply:
I have https://github.com/handsontable/handsontable/pull/10758/files pinned in my TODO list. My idea was to wait until it's merged into `develop`, then switch my development on top of the upstream as well, to have the proper builds running in GH, better visibility etc - I was actually thinking about recreating my PRs on top of upstream, it's just 3 of them by now, I can even do it manually if needed. 

If you agree it makes sense, then I'd not duplicate these changes in my PR and just sync my changes once I'm moving there.

WDYT?

Reply:
Makes sense, agreed